src/ducks/subject.js

```diff
@@ -182,7 +182,9 @@ const prepareForNewSubject = (subject) => {
     dispatch(createClassification(subject));
     dispatch(changeFrame(0));
     if (aggregationWorkflows.indexOf(activeWorkflowID) >= 0) {
-      subject && dispatch(fetchAggregations(subject.id, activeWorkflowID));
+      const aggWorkflowId = activeWorkflowID === config.challengingHebrew ?
+        config.hebrewKeyword : config.arabicKeyword;
+      subject && dispatch(fetchAggregations(subject.id, aggWorkflowId));
     } else {
       dispatch(clearAggregations());
```

Comment:
I just noticed that `fetchAggregations()` doesn't properly reset the `redux.aggregations` store like `clearAggregations()` does - specifically, when fetching a new aggregation, the `aggregations.showHints` doesn't change (which is fine), and `aggregations.keywordWorkflow` retains its old value until we enter a success state. (Which... should be fine.)

This might cause some weird edge case shenanigans, but I don't believe there's anything to worry about, in practice. Still, worth noting it here in case future me is looking for where past me dropped the ball.